src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/AbstractAddItemCommand.cs

```diff
@@ -67,19 +68,53 @@ protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree
+            var addedElements = await OrderingHelper.AddItems(_projectVsServices.ActiveConfiguredProject, _projectVsServices.ProjectLockService, () => OnAddingNodesAsync(nodeToAddTo)).ConfigureAwait(true);
 
-            if (addedNodes.Any())
+            if (addedElements.Any())
             {
-                // Make sure to change ConfigureAwait to "true" if we need switch back.
-                await OnAddedNodesAsync(_projectVsServices.ActiveConfiguredProject, node, addedNodes, updatedNode).ConfigureAwait(false);
+                await new ProjectAccessor(_projectVsServices.ProjectLockService).OpenProjectForWriteAsync(_projectVsServices.ActiveConfiguredProject, project => OnAddedElements(project, addedElements, node, nodeToAddTo)).ConfigureAwait(true);
+
+                // Re-select the node that was the target.
+                await _projectTree.TreeService.PublishLatestTreeAsync(waitForFileSystemUpdates: true).ConfigureAwait(true);
+                HACK_NodeHelper.Select(_projectVsServices.ActiveConfiguredProject, _serviceProvider, node);
+
+                // If the node we wanted to add to is a folder, make sure it is expanded.
+                if (nodeToAddTo.IsFolder)
+                {
+                    HACK_NodeHelper.ExpandFolder(_projectVsServices.ActiveConfiguredProject, _serviceProvider, nodeToAddTo);
+                }
+
+                await Task.Delay(1000).ConfigureAwait(true);
+
+                if (addedElements.Length == 1)
+                {
+                    var unconfiguredProject = _projectVsServices.ActiveConfiguredProject.UnconfiguredProject;
+                    var filePath = unconfiguredProject.MakeRooted(addedElements[0].Include);
+                    var addedNode = _projectTree.CurrentTree.FindImmediateChildByPath(filePath);
+
+                    Assumes.NotNull(addedNode);
+
+                    var uiShellOpenDocument = _serviceProvider.GetService(typeof(SVsUIShellOpenDocument)) as IVsUIShellOpenDocument;
+
+                    var sp = (IOleServiceProvider)_serviceProvider.GetService(typeof(IOleServiceProvider));
+                    var docDataExisting = IntPtr.Zero;
+                    var logicalView = VSConstants.LOGVIEWID.Code_guid;
+                    var openFlags = __VSOSEFLAGS.OSE_ChooseBestStdEditor;
+                    var caption = "%2";  // %2 tells the IDE to call GetProperty(itemid, VSHPROPID_Caption, ...) to get the caption for the window.
+                    Verify.HResult(uiShellOpenDocument.OpenStandardEditor(
+                        (uint)openFlags,
+                        filePath,
+                        ref logicalView,
+                        caption,
+                        (IVsUIHierarchy)unconfiguredProject.Services.HostObject,
+                        addedNode.GetHierarchyId(),
+                        docDataExisting,
+                        sp,
+                        out var windowFrame));
```

Comment:
You can use `out _` for unused out params.

Reply:
Nice!